htlcswitch/interfaces.go

```diff
@@ -138,6 +138,20 @@ type ChannelUpdateHandler interface {
 	// clean. This can be used with dynamic commitment negotiation or coop
 	// close negotiation which require a clean channel state.
 	ShutdownIfChannelClean() error
+
+	// Flush is a method that disables htlc adds to the channel until it has
+	// reached an empty state. When we reach zero HTLCs, the supplied
+	// function will be called.
+	Flush(func()) error
```

Comment:
Will this error out if a caller tries to flush while one is already outstanding? Or does it get added to a queue to permit multiple callers?

Reply:
It error's out right now and I'm reticent to handle it differently. If we wanted to permit multiple callers it would make more sense to have a list of continuations that are invoked in sequence once we hit HTLC-0 but I'm inclined to let a use case where we actually want that pop up first.